common/src/main/java/org/opensearch/sql/common/utils/StringUtils.java

```diff
@@ -108,6 +108,18 @@ public static String format(final String format, Object... args) {
     return String.format(Locale.ROOT, format, args);
   }
 
+  public static String getAliasWithNested(String name) {
```

Comment:
Smart parsing could be very complex, imagine cases like
```
nested(`fieldA`, nested('field.subfield))
nested(`fieldB`, "nested(...)")
```
I think we should smartly use parsing provided by ANTLR engine.

Consider making new tree visitor which would process `nested` only. Something like `HighlightAnalyzer`, I guess.

Reply:
I tested for multiple nested functions within a nested function and it's not supported in legacy. Also, nested in SELECT only supports one argument.

Reply:
We should think 2 steps forward and make a scalable solution considering future changes. Nesting `nested`; `reverse_nested` which string as a second argument; string could contain anything.

Reply:
Good point. I think `reverse_nested` right now is a big unknown. Maybe we can leave this PR for now until we find out more about `reverse_nested`.

Reply:
I actually don't think reverse_nested works. I run the query that is in integration tests and it doesn't work.

`opensearchsql> SELECT COUNT(*) FROM opensearch-sql_test_index_nested_type GROUP BY  nested(message.info),filter('myFilter',message.info = 'a'),reverse_nested(so
.............. meField,'');
OpenSearchException({'error': {'reason': 'There was internal problem at backend', 'details': 'Aggregation type nested is not yet implemented', 'type': 'SqlFeatureNotImplementedException'}, 'status': 500})`

Test can be found in https://github.com/Bit-Quill/opensearch-project-sql/blob/2b918a650fc73bff36ef5e202126e25966fcf105/integ-test/src/test/java/org/opensearch/sql/legacy/AggregationIT.java#L1073-L1093

Reply:
`reverse_nested` also has the argument as the alias. See https://bitquill.atlassian.net/wiki/spaces/~256218836/blog/2023/01/11/2691989567/OpenSearch+SQL+nested+and+reverse+nested+functions#:~:text=reverse_nested%20allows%20us%20to%20do%20that%20%E2%80%93%20SELECT%20SUM(reverse_nested(comment.likes))%20FROM%20nested_objects%20GROUP%20BY%20nested(message.info)%20will%20correctly%20add%20up%20comment.likes%20for%20each%20group%20of%20message.info